### ISSUE TITLE
Handle epoll errors for when handling device events 

### ISSUE THREAD

**Issue by iulianbarbu** [firecracker-microvm/firecracker#1627]
At the moment, the current device event handling implementation, for net, block and vsock devices, does not handle errors which might occur on the epoll registered event `fd`s (`EPOLLHUP`, `EPOLLRDHUP`, `EPOLERR`). We must `unregister` these `fd`s for which we receieved error events.

**Comment by acatangiu** [comment]
I don't think we should unregister in case of any of those events. But all devices do need to gracefully handle them.

**Comment by serban300** [comment]
In the meanwhile we switched to using the EventHandler from rust-vmm. We should check if it handles these events or if we should move this issue there.
